test/byContract/IexecEscrow/IexecEscrowToken.test.ts

```diff
@@ -0,0 +1,473 @@
+                amounts: [standardAmount, standardAmount.mul(2)],
+                targets: [AddressZero, accountB.address],
+            };
+            const depositForArrayArgs = Object.values(depositForArrayParams) as [
+                BigNumber[],
+                string[],
+            ];
+            const depositTotalAmount = getTotalAmount(depositForArrayParams.amounts);
+
+            await rlcInstanceAsAccountA
+                .approve(iexecPoco.address, depositTotalAmount)
+                .then((tx) => tx.wait());
+            await expect(
+                iexecPocoAsAccountA.depositForArray(...depositForArrayArgs),
+            ).to.be.revertedWith('ERC20: mint to the zero address');
+        });
+    });
+
+    describe('Withdraw', () => {
+        it('Should withdraw tokens', async () => {
+            await rlcInstanceAsAccountA
+                .approve(iexecPoco.address, standardAmount)
+                .then((tx) => tx.wait());
+            await iexecPocoAsAccountA.deposit(standardAmount).then((tx) => tx.wait());
+
+            expect(await iexecPocoAsAccountA.callStatic.withdraw(standardAmount)).to.be.true;
+            await expect(iexecPocoAsAccountA.withdraw(standardAmount))
+                .to.changeTokenBalances(
+                    rlcInstance,
+                    [iexecPoco, accountA],
+                    [-standardAmount, standardAmount],
+                )
+                .to.emit(rlcInstance, 'Transfer')
+                .withArgs(iexecPoco, accountA.address, standardAmount)
+                .to.changeTokenBalances(iexecPoco, [accountA], [-standardAmount])
+                .to.emit(iexecPoco, 'Transfer')
+                .withArgs(accountA.address, AddressZero, standardAmount);
+        });
+        it('Should withdraw zero token', async () => {
+            expect(await iexecPocoAsAccountA.callStatic.withdraw(0)).to.be.true;
+            await expect(iexecPocoAsAccountA.withdraw(0))
+                .to.changeTokenBalances(rlcInstance, [iexecPoco, accountA], [-0, 0])
+                .to.emit(rlcInstance, 'Transfer')
+                .withArgs(iexecPoco, accountA.address, 0)
+                .to.changeTokenBalances(iexecPoco, [accountA], [-0])
+                .to.emit(iexecPoco, 'Transfer')
+                .withArgs(accountA.address, AddressZero, 0);
+        });
+        it('Should not withdraw native tokens with empty balance', async () => {
+            await expect(
+                iexecPocoAsAccountA.withdraw(standardAmount),
+            ).to.be.revertedWithoutReason();
+        });
+        it('Should not withdraw tokens with insufficient balance', async () => {
+            await rlcInstanceAsAccountA
+                .approve(iexecPoco.address, standardAmount)
+                .then((tx) => tx.wait());
+            await iexecPocoAsAccountA.deposit(standardAmount).then((tx) => tx.wait());
+
+            await expect(
+                iexecPocoAsAccountA.withdraw(standardAmount.mul(2)),
```

Comment:
To test edges (when possible), amount + 1 instead of amount * 2
(multiple occurrences)

Reply:
[several cleans](https://github.com/iExecBlockchainComputing/PoCo/pull/141/commits/4f91a47f4087fb404718953064ef188b393de48d)